client/src/App.tsx

```diff
@@ -87,10 +87,16 @@ function App() {
     console.debug('[Lean4web] Loading preferences')
 
     let saveInLocalStore = false;
-    let newPreferences: any = { ...preferences } // TODO: need `any` instead of `IPreferencesContext` here to satisfy ts
-    for (const [key, value] of Object.entries(preferences)) {
+    let newPreferences: Record<keyof IPreferencesContext, any> = { ...preferences }
+    for (const [key, value] of (Object.entries(preferences) as Entries<IPreferencesContext>)) {
       let storedValue = window.localStorage.getItem(key)
-      if (storedValue) {
+      // prefer URL params over stored
+      const searchParams = new URLSearchParams(window.location.search);
+      if (preferenceParams.includes(key) && searchParams.has(key)) {
+        const paramValue = !(searchParams.get(key) === "false")  // expecting true or false
```

Comment:
probably here we might need to make a similar test as is done below as we have settings which are either boolean or string:  `if (typeof value === 'boolean')` etc.

Reply:
Unfortunately, all values out of searchparams are string, so you have to do stringly-based checks. What are the possible values we want to support? That's why populating the URL can be helpful, for discovery and so that people don't have to come up with the URL on their own.

Reply:
```suggestion
        if (typeof value === 'string') {
          newPreferences[key] = searchParams.get(key)
        } else if (typeof value === 'boolean') {
          // Boolean values
          newPreferences[key] = !(searchParams.get(key) === "false")
        } else {
          // other values aren't implemented yet.
          console.error(`[Lean4web] Preferences (key: ${key}) contain a value of unsupported type: ${typeof value}`)
        }   
```

That's true, but the same applies to the `storedValue` below: it's always a string. My point is that `value` is the default value for the preference in question and it has the right type. So I thought you could do something like this, can't you?